core/src/main/java/com/amplitude/core/events/BaseEvent.kt

```
@@ -1,14 +1,16 @@
 package com.amplitude.core.events
 
+import com.amplitude.core.platform.plugins.AnalyticsEvent
+
 /**
  * BaseEvent for SDK
  */
-open class BaseEvent : EventOptions() {
-    open lateinit var eventType: String
-    var eventProperties: MutableMap<String, Any?>? = null
-    var userProperties: MutableMap<String, Any?>? = null
-    var groups: MutableMap<String, Any?>? = null
-    var groupProperties: MutableMap<String, Any?>? = null
+open class BaseEvent : EventOptions(), AnalyticsEvent {
+    override lateinit var eventType: String
+    override var eventProperties: MutableMap<String, Any>? = null
```

This is a huge change, I'm not sure if I'm remembering correctly that null values get filtered out. 

Do we need to do this now?

---

yes, it's also how iOs works. I can bring back old function but it will silently drop params in this case